### PR TITLE
Change backup dest credentials type

### PR DESCRIPTION
**Pull request by lsviben**
**What this PR does / why we need it**:
Changes the destination credentials type from `ObjecReference` to `*SecretReference`. Should have been like this from the start, I just noticed this while adding API for credentials.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: cmd/kubermatic-api/swagger.json [hunk 1]
```diff
@@ -17886,7 +17886,7 @@
           "x-go-name": "BucketName"
         },
         "credentials": {
-          "$ref": "#/definitions/ObjectReference"
+          "$ref": "#/definitions/SecretReference"
```
Comment (kron4eg) [inline cmd/kubermatic-api/swagger.json:17889]: @zreigz is our API ready to handle such change?
EDIT:
s/api/dashboard/

Comment (lsviben) [reply cmd/kubermatic-api/swagger.json:17889]: The CRD changes with backup destinations got merged last week(https://github.com/kubermatic/kubermatic/pull/8021)
The API changes with above got merged today. (https://github.com/kubermatic/kubermatic/pull/8228)
Dashboard is not using this yet, and nothing is using this in the API, and the CRD changes are not used anywhere yet as well. 
So its a good moment to change this.

Comment (kron4eg) [reply cmd/kubermatic-api/swagger.json:17889]: :+1:

**Review by kron4eg — APPROVED** [review]
/approve

**Pull request merged**